local_planner/CMakeLists.txt

```diff
@@ -134,9 +150,10 @@ add_library(local_planner
   src/nodes/star_planner.cpp
   src/nodes/planner_functions.cpp
   src/nodes/common.cpp
-  src/nodes/rviz_world_loader.cpp
+  ${RVIZ_WORLD_LOADER}
```

Comment:
Something more like:

```
set(planner_cpp_files a.cpp
                      b.cpp
                      c.cpp)

if ADD_THE_THING
set(planner_cpp_files ${planner_cpp_files}
                      d.cpp)
endif

add_library(local_planner ${planner_cpp_files)

Reply:
The implemented logic is for not loading the rviz world when the arg is used. Hence the proposed solution is not applicable.